From fdfc8089429b58b4ac901926fe83fa85b0b7bfc1 Mon Sep 17 00:00:00 2001
From: San Mehat <san@google.com>
Date: Thu, 17 Sep 2009 14:27:41 -0700
Subject: staging: android: ramconsole: Ensure ramconsole does not get
 cluttered by apanic threads
Patch-mainline: HEAD
Git-commit: fdfc8089429b58b4ac901926fe83fa85b0b7bfc1

[Note, this is part of a patch from Sam, just the drivers/staging/
portion, that adds a function that the apanic code calls, but the apanic
code isn't here, so just include part of this to make merges and diffs
easier and this keeps things self-contained - gregkh]

Signed-off-by: San Mehat <san@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/drivers/staging/android/ram_console.c b/drivers/staging/android/ram_console.c
index 8f18a59..d735be4 100644
--- a/drivers/staging/android/ram_console.c
+++ b/drivers/staging/android/ram_console.c
@@ -146,6 +146,14 @@ static struct console ram_console = {
 	.index	= -1,
 };
 
+void ram_console_enable_console(int enabled)
+{
+	if (enabled)
+		ram_console.flags |= CON_ENABLED;
+	else
+		ram_console.flags &= ~CON_ENABLED;
+}
+
 static void __init
 ram_console_save_old(struct ram_console_buffer *buffer, char *dest)
 {
